app/controllers/result_filters/location_controller.rb

```diff
@@ -53,16 +47,35 @@ def provider_option_selected?
       filter_params[:l] == "3"
     end
 
-    def params_for_provider_search
-      filter_params.except(:lat, :lng, :rad, :loc, :lq)
-    end
-
-    def params_for_across_england_search
-      filter_params.except(:lat, :lng, :rad, :loc, :lq, :query)
+    def get_params_for_selected_option(all_params = nil)
```

Comment:
Is there a reason why we give a default of `nil` here? Feels like stuff is expecting a hash, so i'd expect a default to an empty hash

Reply:
(if a default is even required)